src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -522,50 +524,69 @@ public SCM build(SCMHead head, SCMRevision revision) {
                 }
             }
         }
+        String browserUrl = buildBitbucketClient().getRepositoryUri(repositoryType, BitbucketRepositoryProtocol.HTTP, null, repoOwner, repository);
         if (head instanceof PullRequestSCMHead) {
             PullRequestSCMHead h = (PullRequestSCMHead) head;
             if (repositoryType == BitbucketRepositoryType.MERCURIAL) {
                 MercurialSCM scm = new MercurialSCM(getRemote(h.getRepoOwner(), h.getRepository(),
                         BitbucketRepositoryType.MERCURIAL));
                 // If no revision specified the branch name will be used as revision
-                scm.setRevision(revision instanceof MercurialRevision
-                        ? ((MercurialRevision) revision).getHash()
-                        : h.getBranchName()
-                );
-                scm.setRevisionType(RevisionType.BRANCH);
+                if (revision instanceof MercurialRevision) {
```

Comment:
Ah, probably https://github.com/jenkinsci/mercurial-plugin/pull/82/files#diff-c31f4dd29af8cb912ee0613331c0e3c1L228 in https://github.com/jenkinsci/mercurial-plugin/pull/82 is the explanation: `getKey` will be inappropriate otherwise.